label-external-pull-request/action.yml

```
@@ -0,0 +1,38 @@
+name: Label External Pull Requests
+description: Checks if a PR is a fork and applies an external label accordingly
+
+inputs:
+  github_token:
+    description: The GitHub token to use read member status
+    required: true
+  team:
+    description: The team slug for external collaborators
+    required: true
+  org:
+    description: The organization to check membership in
+    required: true
+  username:
+    description: The username to check membership for
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/github-script@v7
+      id: set-result
+      with: 
+        github-token: ${{ inputs.github_token }}
+        script: |
+          const response = await github.rest.teams.getMembershipForUserInOrg({
+            org: "${{ inputs.org }}",
+            team_slug: "${{ inputs.team }}",
+            username: "${{ inputs.username }}",
+          });
+          return response.data.state === 'active';
```

I don't think it's highly necessary for this specific case, but as a matter of good practice, let's avoid passing user supplied inputs directly and use environment variables instead, they are guaranteed to be treated as strings and avoid any data injection vulnerability.

```suggestion
    - uses: actions/github-script@v7
      id: set-result
      env:
         GITHUB_ORG: ${{ inputs.org }}
         GITHUB_TEAM_SLUG: ${{ inputs.team }}
         GITHUB_USERNAME: ${{ inputs.username }}
      with: 
        github-token: ${{ inputs.github_token }}
        result-encoding: string
        script: |
          const response = await github.rest.teams.getMembershipForUserInOrg({
            org: process.env.GITHUB_ORG,
            team_slug: process.env.GITHUB_TEAM_SLUG,
            username: process.env.GITHUB_USERNAME,
          });
          return response.data.state === 'active';
```

---

makes sense